cosmic_ray/config.py

```diff
@@ -13,14 +13,23 @@ def load_config(filename=None):
     If `filename` is `None`, then configuration gets read from stdin.
 
     Returns: A configuration dict.
+
+    Raises:
+      ConfigError: If there is an error loading the config.
     """
     if filename is None or filename == '-':
         LOG.info('Reading config from stdin')
-        return yaml.safe_load(sys.stdin)
+        try:
+            return yaml.safe_load(sys.stdin)
+        except yaml.parser.ParserError as exc:
+            raise ConfigError() from exc
 
-    with open(filename, mode='rt') as handle:
-        LOG.info('Reading config from %r', filename)
-        return yaml.safe_load(handle)
+    try:
+        with open(filename, mode='rt') as handle:
```

Comment:
This is a text file, so there's an implicit default encoding here. The YAML spec (http://www.yaml.org/spec/1.2/spec.html#id2771184) implies we should use UTF-8 unless we want to support UTF-16 too through explicit byte order marks (I guess not).

Reply:
Apparently PyYAML screws up YAML with BOMs anyway, so we must assume UTF-8.